rs-backend/src/server.rs

```diff
@@ -522,6 +456,30 @@ fn mk_stats(data: &[&TestRun], crate_name: &str, phases: &[String]) -> Value {
         .build()
 }
 
+fn on_push(req: &mut Request) -> IronResult<Response> {
+    // FIXME we are throwing everything away and starting again. It would be
+    // better to read just the added files. These should be available in the
+    // body of the request.
+
+    println!("received onpush hook");
+
+    let mut responder = move || {
+        git::update_repo(git::get_repo_path()?)?;
+
+        println!("updating from filesystem...");
+        let new_data = InputData::from_fs()?;
+
+        let rwlock = req.get::<State<InputData>>().unwrap();
```

Comment:
I'm a bit confused here (in part I think because the names are all very generic - would be good to identify with in the name or a comment what data is being manipulated here). It looks like `data` is the main backing data for the program, in which case why do we `get` it from the `Request`?